models/optional_modules/core_web_vitals/snowflake/snowplow_web_vital_measurements.sql

```diff
@@ -0,0 +1,262 @@
+{{
+  config(
+    enabled=var("snowplow__enable_cwv", false) and target.type == 'snowflake' | as_bool(),
+    sql_header=snowplow_utils.set_query_tag(var('snowplow__query_tag', 'snowplow_dbt'))
+  )
+}}
+
+with by_url_and_device as (
+
+  select distinct
+
+    page_url,
+    device_class,
+    'all' as country,
+    'last {{var('snowplow__cwv_days_to_measure')|string }} days' as time_period,
+    count(*) over (partition by page_url, device_class) as page_view_count,
+    nullif(percentile_cont(0.{{ var('snowplow__cwv_percentile') }}) within group (order by lcp) over (partition by page_url, device_class), 0) as lcp_{{ var('snowplow__cwv_percentile') }}p,
+    percentile_cont(0.{{ var('snowplow__cwv_percentile') }}) within group (order by fid) over (partition by page_url, device_class) as fid_{{ var('snowplow__cwv_percentile') }}p,
+    percentile_cont(0.{{ var('snowplow__cwv_percentile') }}) within group (order by cls) over (partition by page_url, device_class) as cls_{{ var('snowplow__cwv_percentile') }}p,
+    percentile_cont(0.{{ var('snowplow__cwv_percentile') }}) within group (order by ttfb) over (partition by page_url, device_class) as ttfb_{{ var('snowplow__cwv_percentile') }}p,
+    'by_url_and_device' as measurement_type
```

Comment:
Could we support a list of percentile values but only do the final result calculation on the first/last value in the array? Might make for some easier visuals but keeps a single pass/fail result? Or is that not really how people use these values?

Reply:
AFAIK most people would just leave everything as is at the default 75 percentile mark as that is the official way to evaluate the results. Making it a variable just allows people to alter this logic if they want to do so.